### DISCUSSION TITLE
oidc_util_http_call: curl_easy_perform() failed on: https://<<endpoint>>  (getaddrinfo() thread failed to start\n)

### DISCUSSION THREAD

**Discussion by cheewanwoo** [OpenIDC/mod_auth_openidc#977] [General]
does any one seen such problem ?
oidc_util_http_call: curl_easy_perform() failed on: https://<<hostname>>  (getaddrinfo() thread failed to start\n)

where
OIDCProviderMetadataURL https://<<hostname>>/ext/.well-known/openid-configuration

**Comment by zandbelt** [comment]
some googling suggests that it is due to a local issue, firewall- or resource- related

**Comment by cheewanwoo** [comment]
yeah, I have the same idea as well and I had a network resource to trace the network routing and indeed it was a firewall issue and after the firewall issue was fixed, I could see traffic going through to the OIDCProviderMetadataURL https://<>/ext/.well-known/openid-configuration endpoint  by testing it with curl -v https://<hostname>/ext/.well-known/openid-configuration

And I even tried with 
OIDCOutgoingProxy http://a10corp-bcprx.wip.uhc.com:8083  OR
OIDCOutgoingProxy a10corp-bcprx.wip.uhc.com:8083

And the same configuration works in our stage env.  But off course our stage env has a different  OIDCProviderMetadataURL https://<>/ext/.well-known/openid-configuration endpoint

I have looked at curl and libcurl and compared it to the stage env and its all the same installations

Here is the OIDC config
OIDCCABundlePath  /cap/httpd24/config/apache_ca.pem
OIDCHTTPTimeoutShort 10
#OIDCOutgoingProxy [<scheme>://]<host>[:<port>]
OIDCOutgoingProxy http://a10corp-bcprx.wip.uhc.com:8083
OIDCRedirectURI /login/hsid
OIDCCryptoPassphrase <<protected>>
OIDCProviderMetadataURL https://<<hostname>>/ext/.well-known/openid-configuration
OIDCScope "openid profile"
OIDCAuthRequestParams pfidpadapterid=<<id>>
OIDCSSLValidateServer On
OIDCClientID <<id>>
OIDCClientSecret <<secret>>
OIDCCookie CAPSESSION
OIDCSessionCookieChunkSize 0
OIDCStateMaxNumberOfCookies 5 true
OIDCSessionInactivityTimeout 1200
OIDCSessionType client-cookie
OIDCDefaultURL https://<<hostname>>/account
OIDCDefaultLoggedOutURL https://<<hostname>>/logout
OIDCRemoteUserClaim sub
OIDCAuthNHeader UUID
OIDCHTTPTimeoutShort 10
OIDCStateTimeout 900
OIDCPassRefreshToken On
